libvast/vast/system/node.hpp

```diff
@@ -13,20 +13,40 @@
 
 #pragma once
 
+#include <map>
 #include <string>
 
 #include <caf/actor.hpp>
 #include <caf/event_based_actor.hpp>
 #include <caf/stateful_actor.hpp>
 
+#include "vast/aliases.hpp"
 #include "vast/command.hpp"
 #include "vast/filesystem.hpp"
+#include "vast/system/spawn_arguments.hpp"
 #include "vast/system/tracker.hpp"
 
 namespace vast::system {
 
+struct node_state;
+
+using node_actor = caf::stateful_actor<node_state>;
+
 /// State of the node actor.
 struct node_state {
+  // -- member types ----------------------------------------------------------
+
+  /// Spawns a component (actor) for the NODE with given spawn arguments.
+  using component_factory = maybe_actor (*)(node_actor*, spawn_arguments&);
+
+  /// Identifies a command and its parent by name. The two names in conjunction
+  /// uniquely identify a component, e.g., by disambiguating "import pcap" from
+  /// "export pcap".
```

Comment:
Does this approach actually work? If `pcap` had a subcommand `foo`, then we may have `(pcap, foo)` twice in there, once for `import` and once for `export`. It seems that storing the full path `(import, pcap, foo)` needs to be the map key. Or using a recursive structure.

Reply:
It does work "for now", but I see your point. A hierarchical mapping is easier to extend in the future.